src/main/java/com/cloudbees/jenkins/support/impl/RootCAs.java

```diff
@@ -0,0 +1,149 @@
+import java.util.WeakHashMap;
+
+/**
+ * @author schristou88
+ */
+@Extension
+public class RootCAs extends Component {
+
+  private final WeakHashMap<Node, String> certCache = new WeakHashMap<Node, String>();
+
+  @Override
+  public boolean isSelectedByDefault() {
+    return false;
+  }
+
+  @NonNull
+  @Override
+  public Set<Permission> getRequiredPermissions() {
+    return Collections.singleton(Jenkins.ADMINISTER);
+  }
+
+  @NonNull
+  @Override
+  public String getDisplayName() {
+    return "Root CAs";
+  }
+
+  @Override
+  public void addContents(@NonNull Container container) {
+    Jenkins j = Jenkins.getInstance();
+    addContents(container, j);
+    for (Node node : j.getNodes()) {
+      addContents(container, node);
+    }
+  }
+
+  private void addContents(@NonNull Container container, final @NonNull Node node) {
+    Computer c = node.toComputer();
+    if (c == null) {
+      return;
+    }
+    String name;
+    if (node instanceof Jenkins) {
+      name = "master";
+    } else {
+      name = "slave/" + node.getNodeName();
+    }
+    container.add(
+            new Content("nodes/" + name + "/RootCA.txt") {
+              @Override
+              public void writeTo(OutputStream os) throws IOException {
+                PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(os, "utf-8")));
+                try {
+                  out.println(getRootCA(node));
+                } catch (IOException e) {
+                  e.printStackTrace(out);
+                } catch (InterruptedException e) {
+                  e.printStackTrace(out);
+                } finally {
+                  out.flush();
+                }
+              }
+            }
+    );
+  }
+
+  public String getRootCA(Node node) throws IOException, InterruptedException {
+    return AsyncResultCache.get(node, certCache, new GetRootCA(), "Root CA info",
+            "N/A: Either no connection to node, or no cached result");
+  }
+
+
+  private static final class GetRootCA implements Callable<String, RuntimeException> {
+    @edu.umd.cs.findbugs.annotations.SuppressWarnings(
+            value = {"RV_RETURN_VALUE_IGNORED_BAD_PRACTICE", "DM_DEFAULT_ENCODING"},
+            justification = "Best effort"
+    )
+    public String call() {
+      StringWriter writer = new StringWriter();
+      getRootCAList(writer);
+      return writer.toString();
+    }
+
+    private static final long serialVersionUID = 1L;
+  }
+
+  public static void getRootCAList(StringWriter writer) {
+    KeyStore instance = null;
+    try {
+      instance = KeyStore.getInstance(KeyStore.getDefaultType());
+      while (instance.aliases().hasMoreElements()) {
+        String s = instance.aliases().nextElement();
+        writer.append("========");
+        writer.append("Alias: " + s);
+        writer.append(instance.getCertificate(s).getPublicKey().toString());
+        writer.append("Trusted certificate: " + instance.isCertificateEntry(s));
```

Comment:
Not sure I understand. Obtaining the certificate requires the alias, so the last two things should be in the loop.